File: project/boram-onboarding/src/main/java/inner/circle/boram_onboarding/survay/dto/SurveyCreateRequest.java
package inner.circle.boram_onboarding.survay.dto;

import inner.circle.boram_onboarding.survay.enumerated.QuestionType;
import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter @Setter
public class SurveyCreateRequest {
    private String title;
    private String description;
    private List<QuestionRequest> questions;

    // QuestionRequest 내부 클래스
    @Getter @Setter
    public static class QuestionRequest {
        private String name;
        private String description;
        private QuestionType type;
        private Boolean required;
        private List<String> options; // 단일/다중 선택일 때만


    }
}

File: project/jihochoi-onboarding/QueryBox/src/main/kotlin/com/wlghsp/querybox/ui/dto/SurveyCreateRequest.kt
package com.wlghsp.querybox.ui.dto

import com.wlghsp.querybox.domain.survey.Option
import com.wlghsp.querybox.domain.survey.Options
import com.wlghsp.querybox.domain.survey.Question
import com.wlghsp.querybox.domain.survey.QuestionType
import com.wlghsp.querybox.domain.survey.Questions
import com.wlghsp.querybox.domain.survey.Survey

data class SurveyCreateRequest(
    val title: String,
    val description: String,
    val questions: List<QuestionRequest>
) {
    fun toEntity(): Survey {
        val questionEntities = questions.map { it.toEntity() }
        return Survey.of(
            title = title,
            description = description,
            questions = Questions.of(questionEntities.toMutableList())
        )
    }
}

data class QuestionRequest(
    val name: String,
    val description: String,
    val type: QuestionType,
    val required: Boolean,
    val options: List<String>?
) {
    fun toEntity(): Question {
        val optionVOs = options?.map { Option(it) } ?: emptyList()
        val optionsWrapped = if (type.isChoice()) Options(optionVOs) else null
        return Question(name, description, type, required, optionsWrapped)
    }
}